cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java

```diff
@@ -175,6 +178,79 @@ public String cruiseControlHost(String clusterName, String clusterNamespace) {
+    }
+
+    private void testStateWithWrongAnnotation(Vertx vertx, VertxTestContext context, int pendingCalls, CruiseControlEndpoints endpoint, KafkaRebalance kcRebalance, KafkaRebalanceState kafkaRebalanceState) throws IOException, URISyntaxException {
+        MockCruiseControl.setupCCRebalanceResponse(ccServer, pendingCalls, endpoint);
+
+        checkTransitionWithWrongAnnotation(vertx, context,
+                kafkaRebalanceState, KafkaRebalanceState.Ready,
+                KafkaRebalanceAnnotation.approve, kcRebalance)
+                .onComplete(result -> defaultStatusHandler(result, context));
+    }
+
+
+    private Future<KafkaRebalanceStatus> checkTransitionWithWrongAnnotation(Vertx vertx, VertxTestContext context,
+                                                         KafkaRebalanceState currentState,
+                                                         KafkaRebalanceState nextState,
+                                                         KafkaRebalanceAnnotation initialAnnotation,
+                                                         KafkaRebalance kcRebalance) {
+
+        CruiseControlApi client = new CruiseControlApiImpl(vertx, HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS, MockCruiseControl.CC_SECRET, MockCruiseControl.CC_API_SECRET, true, true);
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(true);
+        ConfigMapOperator mockCmOps = supplier.configMapOperations;
+        KafkaRebalanceAssemblyOperator kcrao = new KafkaRebalanceAssemblyOperator(vertx, supplier, ResourceUtils.dummyClusterOperatorConfig()) {
+            @Override
+            public String cruiseControlHost(String clusterName, String clusterNamespace) {
+                return HOST;
+            }
+        };
+
+        Reconciliation recon = new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, CLUSTER_NAMESPACE, RESOURCE_NAME);
+
+        AbstractRebalanceOptions.AbstractRebalanceOptionsBuilder<?, ?> rbOptions = kcrao.convertRebalanceSpecToRebalanceOptions(kcRebalance.getSpec());
+
+        CrdOperator<KubernetesClient,
+                KafkaRebalance,
+                KafkaRebalanceList> mockRebalanceOps = supplier.kafkaRebalanceOperator;
+
+        when(mockCmOps.getAsync(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(Future.succeededFuture(new ConfigMap()));
+        when(mockRebalanceOps.get(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(kcRebalance);
+        when(mockRebalanceOps.getAsync(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(Future.succeededFuture(kcRebalance));
+
+        return kcrao.computeNextStatus(recon, HOST, client, kcRebalance, currentState, initialAnnotation, rbOptions)
+                .compose(result -> {
+                    context.verify(() -> {
+                        assertThat(result.getStatus().getConditions(), StateMatchers.hasStateInConditions(nextState));
+                        System.out.println(result.getStatus());
```

Comment:
Is this a leftover from debugging?

Reply:
Yeah. Thanks for catching this

Reply:
Fixed @kyguy . Also it looks like I didn't removed the test from `KRAO` test class. I have now updated the PR